app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginViewModel.kt

```diff
@@ -73,35 +72,24 @@ class LoginViewModel : BaseViewModel() {
 
     fun login(phone: String, password: String, firebaseToken: String) {
         logD("login: $phone : $password -> $firebaseToken")
-        disposables.add(
-            loginRepository.login(User(phone, password, firebaseToken))
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe({ loginResponse ->
-                    logD("Login successful! Token received!")
-                    onSuccessfulLogin(loginResponse, firebaseToken)
-                }, { throwable ->
-                    firebaseAnalytics.logEvent(Event.LOGIN_FAILED.title, null)
-                    logE("Login failed!", throwable)
-                    onError(throwable)
-                })
+
+        val user = User(
+            user = phone,
+            password = password,
+            fcmToken = firebaseToken
         )
-    }
 
-    private fun registerForNotification(firebaseToken: String) {
-        logD("registerForNotification with $firebaseToken")
-        disposables.add(
-            loginRepository.registerForNotification(firebaseToken)
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe({
-                    logD("Registered for notifications on firebase!")
-                    onSuccessfulRegisteredForNotification()
-                }, { throwable ->
-                    logE("Register for notification failed!", throwable)
+        loginRepository.login(user)
```

Comment:
```suggestion
        loginRepository.login(User(phone, password, firebaseToken))
```
There's no need to use a separate variable for this

Reply:
Done